components/chainhook-sdk/src/observer/http.rs

```diff
@@ -262,99 +234,102 @@ pub fn handle_new_microblocks(
     marshalled_microblock: Json<JsonValue>,
     background_job_tx: &State<Arc<Mutex<Sender<ObserverCommand>>>>,
     ctx: &State<Context>,
-) -> Json<JsonValue> {
-    ctx.try_log(|logger| slog::debug!(logger, "POST /new_microblocks"));
+) -> Result<Json<JsonValue>, Custom<Json<JsonValue>>> {
+    try_info!(ctx, "POST /new_microblocks");
     // Standardize the structure of the microblock, and identify the
     // kind of update that this new microblock would imply
     let chain_event = match indexer_rw_lock.inner().write() {
-        Ok(mut indexer) => {
-            
-            indexer.handle_stacks_marshalled_microblock_trail(
-                marshalled_microblock.into_inner(),
-                ctx,
-            )
-        }
+        Ok(mut indexer) => indexer
+            .handle_stacks_marshalled_microblock_trail(marshalled_microblock.into_inner(), ctx),
         Err(e) => {
-            ctx.try_log(|logger| {
-                slog::warn!(
-                    logger,
-                    "unable to acquire background_job_tx: {}",
-                    e.to_string()
-                )
-            });
-            return Json(json!({
-                "status": 500,
-                "result": "Unable to acquire lock",
-            }));
+            return error_response(
+                format!("Unable to acquire background_job_tx: {}", e.to_string()),
+                ctx,
+            );
         }
     };
 
     match chain_event {
         Ok(Some(chain_event)) => {
             let background_job_tx = background_job_tx.inner();
             match background_job_tx.lock() {
-                Ok(tx) => {
-                    let _ = tx.send(ObserverCommand::PropagateStacksChainEvent(chain_event));
-                }
+                Ok(tx) => match tx.send(ObserverCommand::PropagateStacksChainEvent(chain_event)) {
+                    Ok(_) => {}
+                    Err(e) => {
+                        return error_response(
+                            format!("Unable to send stacks chain event: {}", e.to_string()),
+                            ctx,
+                        );
+                    }
+                },
```

Comment:
Please check the comment above about ignoring success cases and nested statements.